### PR TITLE
santad: Move signature fetching into SNTPolicyProcessor

### PR DESCRIPTION
**Pull request by russellhancox**
This also removes an unnecessary hash and checks code signatures on non-MachO files (which is rare but possible).

It also fixes a rare crash in EndpointSecurityManager

### CODE REVIEW

File: Source/santad/SNTExecutionController.m [hunk 1]
```diff
@@ -115,66 +115,41 @@ - (void)validateBinaryWithMessage:(santa_message_t)message {
...
   // Send the decision to the kernel.
   [self.eventProvider postAction:action forMessage:message];
 
   // Log to database if necessary.
-  if (cd.decision != SNTEventStateAllowBinary &&
-      cd.decision != SNTEventStateAllowCompiler &&
-      cd.decision != SNTEventStateAllowTransitive &&
-      cd.decision != SNTEventStateAllowCertificate &&
-      cd.decision != SNTEventStateAllowScope) {
+  if (!(cd.decision & SNTEventStateAllow)) {
```
Comment (tburgin) [inline Source/santad/SNTExecutionController.m]: We still want to add SNTEventStateAllowUnknown to the event database.

File: Source/santad/SNTPolicyProcessor.m [hunk 2]
```diff
@@ -36,13 +38,27 @@ - (instancetype)initWithRuleTable:(SNTRuleTable *)ruleTable {
...
+- (nonnull SNTCachedDecision *)decisionForFileInfo:(nonnull SNTFileInfo *)fileInfo
+                                        fileSHA256:(nullable NSString *)fileSHA256 {
   SNTCachedDecision *cd = [[SNTCachedDecision alloc] init];
   cd.sha256 = fileSHA256 ?: fileInfo.SHA256;
-  cd.certSHA256 = certificateSHA256;
+
+  // If the binary is a critical system binary, don't check its signature.
+  // The binary was validated at startup when the rule table was initialized.
+  SNTCachedDecision *systemCd = self.ruleTable.criticalSystemBinaries[cd.sha256];
+  if (systemCd) {
```
Comment (tburgin) [inline Source/santad/SNTPolicyProcessor.m]: nit: if (systemCd) return systemCd;

File: Source/santad/SNTExecutionController.m (hunk 1 above)
Comment (russellhancox) [reply Source/santad/SNTExecutionController.m]: Ahh, that's why we didn't use the mask

File: Source/santad/SNTPolicyProcessor.m (hunk 2 above)
Comment (russellhancox) [reply Source/santad/SNTPolicyProcessor.m]: Done

**Pull request merged**